src/gui_common/charts/line/LineChartData.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections.Generic;
+using Godot;
+
+/// <summary>
+///   Dataset to be visualized on the line chart
+/// </summary>
+public class LineChartData
+{
+    private Color lineColor;
+    private bool draw = true;
+
+    // ReSharper disable once CollectionNeverUpdated.Global
+    public List<DataPoint> DataPoints { get; set; } = new List<DataPoint>();
+
+    /// <summary>
+    ///   The icon on the chart legend
+    /// </summary>
+    public Texture IconTexture { get; set; }
+
+    public float LineWidth { get; set; } = 1.3f;
+
+    /// <summary>
+    ///   Used to differentiate the data set's visual by color
+    /// </summary>
+    public Color DataColor
+    {
+        get => lineColor;
+        set
+        {
+            lineColor = value;
+
+            if (DataPoints.Count <= 0)
+                return;
```

Comment:
I don't think this and the other Count check are needed. If the list is empty the foreach just doesn't run.